### ISSUE TITLE
torch.argmax cannot backpropogate

### ISSUE THREAD

**Issue by vvincenttttt** [lanlan96/ARM3D#1]
Hi, nice work!
I saw in your code you use torch.argmax to get idx_obj. however, torch.argmax cannot backpropogate, do you notice it?
thanks!

**Comment by lanlan96** [comment]
Hi, the operation "torch.argmax" is non-differentiable, and it uses the results of objectness of proposals. We calculate the loss of objectness of all proposals and backpropagate the loss. It means that the objecntness module can only learn from the  objectness loss. The following operations based on selection of proposals don't influence the training of objecntness module.
